frontend-shared/README.md

```diff
@@ -0,0 +1,37 @@
+# Shared resources for Hypothesis front-end applications
+
+A package of resources for Hypothesis front-end applications.
+
+#### Requirements
+
+- preact
```

Comment:
adding preact to `peerDependencies` does not work. In every case I tried, it also installed preact and once it did that, hooks were unable to work correctly because they could not import preact modules back --possible due to some module mismatch standard.

Reply:
The problem with `peerDependencies` occurs when:

1. You run `npm install` in the `frontend-shared/` directory and that locally installs `node_modules/preact`.
2. When bundling the bundler does not de-duplicate the modules and two copies of Preact end up in the bundle, with some code using the main copy and other code using the copy from `frontend_shared/node_modules/preact`
3. Having multiple copies of Preact's hooks module in a bundle breaks because the hooks API inherently involves global state: the `useXYZ` functions need to know what the currently-rendering component is. This flaw is not unique to Preact. It affects [React hooks](https://github.com/facebook/react/issues/13991) as well.

For local development of the client, everything works as long as you haven't run `npm install` inside the `frontend-shared/` directory. If you add `@hypothesis/frontend-shared` as a dependency to the client's `package.lock` with the version set to `file:./frontend-shared` then I think yarn should install any runtime dependencies of that project when running `yarn install` for the client.

An alternative solution would be to tell the bundler not to include multiple copies of the library even if they are present in the tree. I don't know off-hand how to do this out of the box with Browserify.